src/View/mainpage/DrawROIWindow/UIDrawROIWindow.py

```diff
@@ -808,6 +792,25 @@ def onDrawClicked(self):
         """
         Function triggered when the Draw button is pressed from the menu.
         """
+        logging.debug("onDrawClicked started")
+        if hasattr(self, 'drawingROI'):
+            if self.drawingROI is not None:
+                self.is_drawing.connect(self.drawingROI.set_is_drawing)
+                self.is_drawing.emit(True)
+                self.is_drawing.disconnect(self.drawingROI.set_is_drawing)
```

Comment:
Interesting, any reason why you have chosen this way?

Reply:
This is found in both the on fill and on draw clicked functions. The reason this was done was to connect the signal to the right drawingROI object since this reference will change in between slice changes. If this were to connect once to one drawingROI object and then the user changes slice this signal would still point to that object. Therefore, it currently connects to the reference, emits the signal, and disconnects from the reference.

Reply:
awesome, good catch too